app/collections/collections-manager.jsx

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import CollectionsCreateForm from './create-form';
+import CollectionSearch from './collection-search';
+
+class CollectionsManager extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      collections: [],
+      error: null,
+      hasCollectionSelected: false
+    };
+
+    this.addToCollections = this.addToCollections.bind(this);
+  }
+
+  addToCollections() {
+    const collections = this.search.getSelected();
+    if (!(collections.length > 0)) return;
+
+    const promises = collections.map(collection =>
+      collection.collection.addLink('subjects', this.props.subjectIDs)
+    );
+
+    Promise.all(promises)
+      .then(() => {
+        this.props.onSuccess();
+      }).catch(error => this.setState({ error }));
+  }
+
+  render() {
+    return (
+      <div className="collections-manager">
+        <h1>Add Subject to Collection</h1>
+
+        <div>
+          {this.state.error &&
+            <div className="form-help error">{this.state.error.toString()}</div>}
```

Comment:
There was a comment on the original issue about improved error messaging. Was that in reference to how the message is displayed? Maybe it's coming in a separate PR.